entity-server-api/src/main/java/org/terracotta/entity/ServiceProvider.java

```diff
@@ -22,46 +22,62 @@
-   * Get an instance of service from the provider.
-   *
-   * @param consumerID The unique ID used to name-space the returned service
-   * @param configuration Service configuration which is to be used
-   * @return service instance
+   * Get an instance of a service from the provider.
+   * Note that this method can be called any number of times, over the life of a ServiceProvider instance, potentially
+   *  concurrently or multiple times for the same consumerID.  If the implementation wishes to intern, or otherwise track,
+   *  the instances it returns, it will need to ensure its approach is robust against concurrent calls.
+   * Note that the consumerID is a unique number for each entity instance (consumerID 0 represents the platform, itself) which
+   *  is used to effectively name-space the service instances under the entities.  An entity instance will have the same
+   *  consumerID for its entire life (including after restart or fail-over).
+   * 
+   * @param consumerID The unique ID used to name-space the returned service for a specific consuming entity (or the
+   *  platform).
+   * @param configuration Service configuration which is to be used.
+   * @return The service instance (which may be either new or a cached/interned instance for this consumerID).
    */
   <T> T getService(long consumerID, ServiceConfiguration<T> configuration);
 
   /**
    * Since a service provider can know how to build more than one type of service, this method allows the platform to query
-   * the extent of that capability.  Returned is a collection of service types which can be returned by getService.
+   * the extent of that capability.  Returned is a collection of service types which can be returned by future getService
+   *  calls.
    *
    * @return A collection of the types of services which can be returned by the receiver.
    */
   Collection<Class<?>> getProvidedServiceTypes();
 
   /**
-   * Clears up state for this ServiceProvider including any persisted state
-   *
-   * Generally platform calls this method during platform initialization so there won't be any entities using
-   * underlying services but ServiceProviders need to make sure that no entities (with the exception of platform whose consumerID is 0) using underlying services
-   *
-   * If there are any failures when clearing state, this method should inform Platform by throwing {@link ServiceProviderCleanupException}
+   * Clears up state for this ServiceProvider including any persisted state.
+   * This is called, by the platform, when it wishes to explicitly clear the server's state (both in-memory and persistent).
+   * Generally, the platform calls this method during platform initialization so there won't be any entities using the
+   *  underlying service instances.  In a more concrete sense, no services instances created before this call are considered
+   *  valid after the call, except for those created for consumerID 0 (which is the platform-internal consumer).
```

Comment:
Wait, does that mean that clear should not clear state linked to `consumerID 0`?